templates/upstart.conf.j2

```diff
@@ -0,0 +1,14 @@
+description "Gunicorn upstart script for {{ gunicorn_app_name }}"
+
+{% if develop or test -%}
```

Comment:
These are the most common `start on` settings internally, but I think it might be better to default to one (maybe the `local-filesystems...` one) and use the other in the example (`vagrant-mounted`). Alternatively, this could be something that we drop from the role and force the service definition to be created by the user.